src/protocol/packets/c2s.rs

```diff
@@ -8,7 +8,17 @@ pub mod handshake {
     def_struct! {
         Handshake {
             protocol_version: VarInt,
-            server_adddress: BoundedString<0, 255>,
+            server_address: BoundedString<0, 255>,
+            server_port: u16,
+            next_state: HandshakeNextState,
+        }
+    }
+
+    def_struct! {
+        BungeecordHandshake {
```

Comment:
Feel free to remove the `BoundedString` from `Handshake` and check the bounds later. Then you can remove `BungeecordHandshake`.